### ISSUE TITLE
math doesn't make sense for monitor positioning with variable size displays and varying scales including fractional scaling

### ISSUE THREAD

**Issue by andar1an** [hyprwm/Hyprland#8241]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

No

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch build-alpine at commit 4520b30d498daca8079365bdb909a8dea38e8d55 dirty (with patches:).
Date: 
Tag: v0.44.1, commits: 
built against aquamarine 0.4.2


flags: (if any)


System Information:
System name: Linux
Node name: tor-dev-01
Release: 6.6.58-1-lts
Version: #2-Alpine SMP PREEMPT_DYNAMIC 2024-10-23 16:48:22


GPU information: 
00:02.0 VGA compatible controller [0300]: Intel Corporation CometLake-U GT2 [UHD Graphics] [8086:9b41] (rev 02) (prog-if 00 [VGA controller])


os-release: NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.21.0_alpha20240923
PRETTY_NAME="Alpine Linux edge"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://gitlab.alpinelinux.org/alpine/aports/-/issues"


plugins:

======Config-Start======

################
### MONITORS ###
################

# See https://wiki.hyprland.org/Configuring/Monitors/
# monitor = name, resolution, position, scale
# recommended rule for random monitor hot-plug: monitor = , preferred, auto, 1
monitor = eDP-1, 2560x1440@60, 0x0, 2
monitor = HDMI-A-1, 3840x2160@30, 0x-1730, 1.25
```

</details>

### Description

I am not sure if what I am noticing is an indication of bad math in Hyprland, or lack of understanding the documentation on my end. 

The documentation states: 
```
Monitors are positioned on a virtual “layout”.
The position is the position, in pixels, of said display in the layout. 
(calculated from the top-left corner)
```

I want to place display HDMI-A-1 above eDP-1. Based on the documentation stating that measurements are made from top-left corners with inverse y Cartesian axis, I would first assume that both my monitors top left corners are (0,0). 

I then move my second display above the (0,0) of the first display by multiplying 2160 * 1.25, and setting the position as `0x-2700`. This is very wrong. 

Through trial and error the actual value should be 0x-1730. The closest I get to this is (1440 + (2880-2700) + 90)=1710 which makes no sense and uses redundant values that I shouldn't use.

The only thing I can think of is that either you are actually measuring from the center, or the scaling of the primary monitor does not maintain the (0,0) at the top left corner of said monitor. 

If I correct for this, and do 2700 + or - (1440/2) for a position of 0x-3420 or 0x-1980, that is also incorrect. 

If I instead decide to do math based on the (0,0) position as the center of the monitors, instead of the top left: Height after scaling for monitor A is 2880, height after scaling for monitor B is 2700, so I need to move B 2700 + difference of one side which is 90. 0x-2790 also doesn't work.

If someone could please point me to where in the code the math is done, that would be appreciated. 

### How to reproduce

Try to position monitors with fractional scaling for varying scales.

**Comment by vaxerski** [comment]
> multiplying 2160 * 1.25

_dividing_... 2160 / 1.25

assuming both are 4K (2160p?) and 1.25 scale:

top monitor: 0,0
bottom monitor: 0,2160/1.25

**Comment by vaxerski** [comment]
fwiw try `wdisplays` for a graphical tool to do this

**Comment by andar1an** [comment]
I just wanted to understand the math. Because it doesn't seem to make sense which may mean problems for development in future. 

The working value is 0x-1730, so none of this math adds up.

**Comment by andar1an** [comment]
In my mind an anchor of (0, 0) stays at same point on primary display even after scaling. It shouldn't be too complex if that is the case. Just a matter of defining primary display which I don't see an option for?

**Comment by andar1an** [comment]
> fwiw try `wdisplays` for a graphical tool to do this

I like it in config, this doesn't change for me. Will definitely look into though, ty for tip. And loving hyprland so far!

**Comment by vaxerski** [comment]
position is top left of the display
its logical size is (pixel res / scale)

thats about it, you're just positioning rectangles

**Comment by andar1an** [comment]
> position is top left of the display
> its logical size is (pixel res / scale)
> 
> thats about it, you're just positioning rectangles

Exactly, that's why it is weird that the math doesn't work. Multiplying or dividing scale, or measuring from center. 

I am pretty confident in my abilities with this type of math haha.

Assuming both displays start with top left corners at same position that math would mean a translation of  about 3000px up, so 0x-3000. But i get 0x-1730 as working value. That is also why I tried to do math based on a detached origin, or from center. 

Not worth worrying much about, I created this more so out of concern in case something may be fishy.
